### PR TITLE
Adding fallback for all non-supported platforms

### PR DESCRIPTION
**Pull request by diamondburned**
Without this, the app would instantly segfault due to `nil` pointer. With this, it errors out properly.

### CODE REVIEW

**Comment by szuecs** [comment]
@diamondburned Thanks for your PR!

Can you please:
1) sign DCO (aka `git commit -s --ammend`) and do a force push ?
2) add maybe a freebsd target if possible to https://github.com/zalando/go-keyring/blob/master/.travis.yml#L4

I am not sure if this change will work for GOOS freebsd. I tried to test it, but it seemed not to work for me:

```
% GOOS=freebsd go build .                                                                                                                          master
# github.com/godbus/dbus
../../godbus/dbus/transport_unix.go:51:3: cannot use t (type *unixTransport) as type transport in return argument:
        *unixTransport does not implement transport (missing SendNullByte method)
../../godbus/dbus/transport_unix.go:57:3: cannot use t (type *unixTransport) as type transport in return argument:
        *unixTransport does not implement transport (missing SendNullByte method)
```

**Comment by diamondburned** [comment]
Before doing those 2, I can only say it works because a friend of mine tested it in a jail. It would simply panic without that, as `freebsd` seems to match into `unix`, leading to a crash by a `nil` pointer dereference. I can't cross-compile from a Linux system because of that same error.

**Comment by diamondburned** [comment]
Latest commit does not panic on any non-supported platforms.

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

File: keyring_fallback.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package keyring
+
+import (
+	"errors"
+	"runtime"
+)
+
+// All of the following method errors out on unsupported platforms
+var errUnsupportedPlatform = errors.New("Unsupported platform: " + runtime.GOOS)
```
Comment (timbunce) [inline keyring_fallback.go]: It would be helpful if this was exporter so users could check for this specific error.

Comment (mikkeloscar) [reply keyring_fallback.go]: Would you mind creating a PR for this? :)

Comment (timbunce) [reply keyring_fallback.go]: Sure, #38.

**Pull request merged**